### DISCUSSION TITLE
TPI in AC Mode

### DISCUSSION THREAD

**Discussion by razorback91** [jmcollin78/versatile_thermostat#474] [Q&A]
Hello,

What are the TPI value used for AC mode ? When I do the calculation by hands it work well for heating but in AC mode I don't have same result.
What can you help to underdand the TPI coeff and if we can change them . I did some math and it seems to be 0.28 and -0.76 but there are very different from the values I defined in config

Thank you

**Comment by jmcollin78** [comment]
Hello, I have seen other messages on this topic. Is it closed now ?

TPI coefs and calculation are done exactly with the same calculation formula. BUt I have to admit it have been done and tested for heating. Maybe the results are not so good for cooling.

**Reply by razorback91** [reply]
I closed the other because when restoring default coeff the entity switch was working. However when I changed the TPI coeff values the power calculation doesn't seem to be good.
I set them to: 0.4771 and 0.07907

and when I do the math :  0.4771 * (temp_target - temp) + 0.07907 * (temp_target - temp_ext)

It doesn't give same results.

**Reply by jmcollin78** [reply]
Can you give me the attributes of your VTherm (Dev tools / States) ?

**Comment by razorback91** [comment]
hvac_modes: heat, cool, off
min_temp: 7
max_temp: 35
target_temp_step: 0.1
friendly_name: Bureau Versatile Thermostat
supported_features: 401
hvac_action: cooling
preset_modes: none, eco, comfort, boost
temperature: 23
preset_mode: none
is_on: true
hvac_mode: cool
is_controlled_by_central_mode: false
frost_temp: 15
eco_temp: 15.5
boost_temp: 7.2
comfort_temp: 19
frost_away_temp: 0
eco_away_temp: 0
boost_away_temp: 0
comfort_away_temp: 0
target_temperature_step: 0.1
ac_mode: true
saved_preset_mode: none
saved_target_temp: 24
window_state: off
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: binary_sensor.ouverture_bureau_ouverture_bureau_any
window_delay_sec: 120
window_auto_enabled: false
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0
security_default_on_percent: 0
last_temperature_datetime: 2024-06-30T17:46:23.569450+02:00
last_ext_temperature_datetime: 2024-06-30T17:48:12.625826+02:00
security_state: false
minimal_activation_delay_sec: 1
device_power: 1
mean_cycle_power: 1
total_energy: 4.08
last_update_datetime: 2024-06-30T17:48:12.971597+02:00
timezone: Europe/Paris
temperature_unit: °C
is_device_active: true
is_used_by_central_boiler: false
is_over_switch: true
is_inversed: false
keep_alive_sec: 0
underlying_switch_0: input_boolean.radiateur_bureau_fake
on_percent: 1
power_percent: 100
on_time_sec: 900
off_time_sec: 0
cycle_min: 15
function: tpi
tpi_coef_int: 0.4771
tpi_coef_ext: 0.07907
current_temperature: 23.8
ext_current_temperature: 23
ema_temp: 23.8

**Reply by jmcollin78** [reply]
Huum, when looking at the code I see something weird indeed.

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/releases/tag/6.2.8

Please give me some feedback because this configuration (TPI with AC Mode is not common and I can't test it on a real environment).

Thabk you for your feedback.

**Reply by razorback91** [reply]
it seems ok but I'll see more if sun decides to show his face
Thank you
